openedx/openedx-platform
# [OC-4409] Browser-Based CSV Viewer for Reports

Adds a purely static CSV viewer to allow viewing CSV reports in the browser. Also adds a link to preview reports generated in the data download page of the instructor dashboard. 

**Project details**:
This PR is part of the "Improved Problem Response Report" project, funded by Australian National University (ANU) and developed by OpenCraft

**JIRA tickets**: OSPR-2424 / OC-4409

**Discussions**: https://edxchange.opencraft.com/t/xblock-reporting-tool/191

**Screenshots**: 
Live preview link:
https://pr18231.sandbox.opencraft.hosting/csv-viewer?csvUrl=https://data.cityofnewyork.us/api/views/25th-nujf/rows.csv?accessType=DOWNLOAD
![screen shot 2018-05-22 at 19 25 37](https://user-images.githubusercontent.com/118837/40367009-1025730a-5df6-11e8-9fd2-8e42c32069d2.png)

**Sandbox URL**: 
* LMS: https://pr18231.sandbox.opencraft.hosting/
* Studio: https://studio-pr18231.sandbox.opencraft.hosting/

**Merge deadline**: "None" 

**Testing instructions**:

1. Visit /csv-viewer?csvUrl=URL_TO_CSV_FILE
2. You should see a sortable, filterable table that displays the contents of the CSV file for viewing in the browser.
OR.
1. Visit the download page in the instructor dashboard.
2. Generate a report.
3. Click on the "Preview" button in the task list.
4. The report should open in the browser for preview.  

Example url: https://pr18231.sandbox.opencraft.hosting/csv-viewer?csvUrl=https://data.cityofnewyork.us/api/views/25th-nujf/rows.csv?accessType=DOWNLOAD 

**Author notes and concerns**:

To view reports stored on S3, SWIFT or some other remote storage, it will need to have CORS set up. 

**Reviewers**
